src/nyc_trees/apps/home/urls.py

```diff
@@ -17,11 +22,20 @@
     url(r'^jobs/(?P<job_id>\d+)/$', r.retrieve_job_status,
         name='retrieve_job_status'),
 
-    url(r'^training/$', r.training_list_page, name="training_list_page"),
-
-    url(r'^training/groups_to_follow/$', r.groups_to_follow),
-
-    # "training" instead of "training/" because the flatpages admin interface
-    # insists that the "URL" (really a URL segment) start with a leading slash
-    url(r'^training(?P<url>.*/)$', include('django.contrib.flatpages.urls')),
+    url(r'^training/$', **training_summary.pure_kwargs()),                                             # NOQA
+    url(r'^training/pure/getting_started/', **getting_started.pure_kwargs()),                          # NOQA
+    url(r'^training/pure/the_mapping_method/', **the_mapping_method.pure_kwargs()),                    # NOQA
+    url(r'^training/pure/tree_data/', **tree_data.pure_kwargs()),                                      # NOQA
+    url(r'^training/pure/tree_surroundings/', **tree_surroundings.pure_kwargs()),                      # NOQA
+    url(r'^training/pure/groups_to_follow/$', **groups_to_follow.pure_kwargs()),                       # NOQA
+    url(r'^training/progress/the_mapping_method/', **the_mapping_method.previous_step.mark_kwargs()),  # NOQA
+    url(r'^training/progress/tree_data/', **tree_data.previous_step.mark_kwargs()),                    # NOQA
+    url(r'^training/progress/tree_surroundings/', **tree_surroundings.previous_step.mark_kwargs()),    # NOQA
+    url(r'^training/progress/groups_to_follow/$', **groups_to_follow.previous_step.mark_kwargs()),     # NOQA
+    url(r'^training/progress/$', **training_summary.previous_step.mark_kwargs()),                      # NOQA
```

Comment:
No, we need dynamically generated `name='<step_name>_pure'` and `name='<step_name>_mark_progress'` kwargs as well, because the steps will look them up.